arch/x86/kvm/vmx/pkvm_init.c

```diff
@@ -465,8 +465,31 @@ static inline void enable_feature_control(void)
 
 static noinline int local_deprivilege_cpu(void)
 {
-	/* TODO */
-	return -EINVAL;
+	int ret;
+
+	asm volatile(
+		"pushfq\n"
+		"popq %%rax\n"
+		"movq %3, %%rdx\n"
+		"vmwrite %%rax, %%rdx\n"
+		"movq %%rsp, %%rax\n"
+		"movq %4, %%rdx\n"
+		"vmwrite %%rax, %%rdx\n"
+		"movq $host_vm_entry_point, %%rax\n"
+		"movq %1, %%rdx\n"
+		"vmwrite %%rax, %%rdx\n"
+		"movl $0, %0\n"
+		"vmlaunch\n"
+		/* successfully deprivileged (CF=0 & ZF=0) */
```

Comment:
If vmlaunch was successful, shouldn't we already be at `host_vm_entry_point: nop` (since it was programmed in GUEST_RIP), not here?

Reply:
Indeed. It is not necessary to have below "ja" instruction. Will update this.